### PR TITLE
Simplifying C# HTTP trigger template

### LINKED ISSUE #63: HTTP C# template is too complicated
I do not think the current code looks approachable. This template probably needs to be rewritten from scratch.

### PR DESCRIPTION
**Pull request by fabiocav**
Resolves #63

### CODE REVIEW

**Comment by davidebbo** [comment]
Very nice! I'd take

```c#
    // Set name to query string, body data, falling back to some default
    name = name ?? data?.name ?? "Azure Functions";

    return req.CreateResponse(HttpStatusCode.OK, "Hello " + name);
```

**Comment by fabiocav** [comment]
I tried to keep the same functionality we had before (and match Node), but I like that!

**Comment by paulbatum** [comment]
This is definitely better, but I think I want to go a step further: is this functionality of asking for a query string parameter really what we want in the default template? I think I would prefer for both this template and the JS template to just return a 200 with a JSON response { message: "hello world" }

**Comment by mathewc** [comment]
I think we go back and forth in these templates between trying to keep them clean/simple and trying to hint at some of the things people can do (e.g. showing APIs).

Whichever way we go, I think we should be consistent across all the templates. I'd be fine if we made them all super clean - just show how to access the input and write/return output, and leave it up to the user to refer to docs to learn about APIs/possibilities.

**Comment by paulbatum** [comment]
OK lets go with Fabio's updated version for now, we can always revisit again later. :shipit:

File: Templates/HttpTrigger-CSharp/run.csx [hunk 1]
```diff
@@ -1,44 +1,22 @@
...
-            Content = new StringContent("Hello " + name)
-        };
-    }
+    // Set name to query string or body data
+    name = name ?? data?.name;
 
-    return Task.FromResult(res);
-}
+    return name == null
+        ? req.CreateResponse(HttpStatusCode.BadRequest, "Please pass  a name on the query string or in the request body")
```
Comment (mathewc) [inline Templates/HttpTrigger-CSharp/run.csx]: Looks like there is an extra space between "pass" and "a"?

**Comment by fabiocav** [comment]
Good eye! This was a simple string fix, so I just merged the PR.

File: Templates/HttpTrigger-CSharp/run.csx (hunk 1 above)
Comment (fabiocav) [inline Templates/HttpTrigger-CSharp/run.csx]: Good eye! This was a simple string fix, so I just merged the PR.

Comment (mathewc) [inline Templates/HttpTrigger-CSharp/run.csx]: Thanks :)

**Pull request merged**